File: euler/39.cc
#include <iostream>

using namespace std;

int solutions( int p )
{
	int ret = 0;
	for( int a = 1; a < p / 2; ++ p )
	{
		for( int b = 1; b < a; ++b )
		{
			for( int c = 1; c <= b; ++ c)
			{
				if( a + b + c == p && ( a*a == b*b+c*c ) )++ret
			}
		}
	}
	return ret;
}

int main( int argc, char ** argv )
{
	int m = 0;
	int pm;
	for( int p = 1; p <= 1000; ++p )
	{
		int t = solutions( p );
		if( m < t )
		{
			m = t;
			pm = p;
		}
	}
	cout<<pm<<endl;
	
}

File: euler/39.py
def solutions( p ):
	ret = 0
	for a in xrange( 1, p /2 ):
		for b in xrange( 1, a+1):
			for c in xrange( 1, b+1 ):
				if a + b + c == p and a*a==b*b+c*c:
					ret = ret + 1
		#			ret.append((a,b,c))
	return ret

mp = 0
mpm = 0
for p in xrange( 1, 1001):
	print p
	t = solutions(p)
	if t > mpm:
		mpm = t
		mp = p

print mp
		
